### PR TITLE
Reorder broker docs; add reference to gcp broker

### PR DESCRIPTION
**Pull request by liu-cong**
Fixes #issue-number or description of the problem the PR solves

## Proposed Changes 

- Move broker related docs into a new folder for better organizing.
- Also add a link to GCP broker
-

cc @grantr @vaikas

### CODE REVIEW

File: docs/eventing/broker/alternate/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+In the Knative Eventing ecosystem, vendors are welcome to bring their own implementation of Broker
```
Comment (vaikas) [inline docs/eventing/broker/alternate/README.md:1]: While I totes agree that vendors are welcome, so are other communities not strictly restricted to vendors. I'd prefer to rework the wording to be more inclusive.

Comment (liu-cong) [reply docs/eventing/broker/alternate/README.md]: Good point! Reworded a little bit.

**Comment by liu-cong** [comment]
BTW, this is how it will look like in docs website:

Screen Shot 2020-05-12 at 4 35 22 PM

**Pull request merged**